### PR TITLE
create open and close buttons, check for valid roomState before saving r...

### PR DESCRIPTION
**Pull request by garronmichael**
...oom

### CODE REVIEW

File: server/rooms/roomController.js [hunk 1]
```diff
@@ -106,9 +106,16 @@ module.exports.save = function(req, res) {
...
+  var endTime = req.body.endTime;
 
+  // if the time is invalid, simply end the function and return
+  if(roomState(startTime, endTime) === 'complete') {
+    res.status(501).send('The room state is invalid for saving');
+    return;
+  }
   // find room and update data
-  Room.findOneAndUpdate({_id: roomId}, {canvas: canvas, text: text}, {upsert: true},
+  Room.findOneAndUpdate({_id: roomId}, {canvas: canvas, text: text, notes: notes, start_time: startTime, end_time: endTime}, {upsert: true},
```
Comment (mcdonnell88) [inline server/rooms/roomController.js]: notes is no longer in the room model; take this out.

**Comment by mcdonnell88** [comment]
overall looks good

**Comment by garronmichael** [comment]
Please hold off this merge, I found something I want to change.

File: server/rooms/roomController.js [hunk 2]
```diff
@@ -106,9 +106,16 @@ module.exports.save = function(req, res) {
   var roomId = req.body.roomId;
   var canvas = req.body.canvas;
   var text = req.body.textEditor;
+  var startTime = req.body.startTime;
+  var endTime = req.body.endTime;
 
+  // if the time is invalid, simply end the function and return
+  if(roomState(startTime, endTime) === 'complete') {
```
Comment (garronmichael) [inline server/rooms/roomController.js]: This logic is bad. It actually prevents me from saving the room as closed. Will need to refactor so closing the PR.

**Pull request closed without merging**